lib/spack/spack/environment.py

```diff
@@ -1180,26 +1183,30 @@ def _add_concrete_spec(self, spec, concrete, new=True):
 
     def install_all(self, args=None):
         """Install all concretized specs in an environment."""
-        for concretized_hash in self.concretized_order:
-            spec = self.specs_by_hash[concretized_hash]
-
-            # Parse cli arguments and construct a dictionary
-            # that will be passed to Package.do_install API
-            kwargs = dict()
-            if args:
-                spack.cmd.install.update_kwargs_from_args(args, kwargs)
-
-            self._install(spec, **kwargs)
-
-            if not spec.external:
-                # Link the resulting log file into logs dir
-                build_log_link = os.path.join(
-                    self.log_path, '%s-%s.log' % (spec.name, spec.dag_hash(7)))
-                if os.path.lexists(build_log_link):
-                    os.remove(build_log_link)
-                os.symlink(spec.package.build_log_path, build_log_link)
-
-        self.regenerate_views()
+        with spack.store.db.read_transaction():
+            for concretized_hash in self.concretized_order:
+                spec = self.specs_by_hash[concretized_hash]
+
+                # Parse cli arguments and construct a dictionary
+                # that will be passed to Package.do_install API
+                kwargs = dict()
+                if args:
+                    spack.cmd.install.update_kwargs_from_args(args, kwargs)
+
+                self._install(spec, **kwargs)
+
+                if not spec.external:
+                    # Link the resulting log file into logs dir
+                    build_log_link = os.path.join(
+                        self.log_path, '%s-%s.log' % (
+                            spec.name, spec.dag_hash(7)
+                        )
+                    )
```

Comment:
Minor. Instead of making this 5 lines (I assume because of flake8) can we write it as:
```python
log_name = '%s-%s' % (spec.name, spec.dag_hash(7))
build_log_link = os.path.join(self.log_path, log_name)
```
?

Reply:
done